### PR TITLE
OEL-779: Review Patterns: Nav, Offcanvas, Pagination, Progress, Table, Timeline

### PR DESCRIPTION
**Pull request by escuriola**
Jira issue(s):
- https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-779

### CODE REVIEW

File: templates/patterns/nav/nav.ui_patterns.yml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
       label: 'Display as tabs'
-      description: 'When enabled, the menu items are displayed as tabs'
+      description: 'When enabled, the menu items are displayed as tabs.'
       preview: true
     vertical:
       type: boolean
       label: 'Vertical'
-      description: 'When enabled, the menu is displayed vertically instead of horizontally'
+      description: 'When enabled, the menu is displayed vertically instead of horizontally.'
       preview: false
```
Comment (abel-santos-corral) [inline templates/patterns/nav/nav.ui_patterns.yml]: Not necessary, default value is false.

File: templates/patterns/nav/nav.ui_patterns.yml [hunk 2]
```diff
@@ -25,25 +25,25 @@ nav:
...
-        center: center
-        end: end
+        start: 'start'
+        center: 'center'
+        end: 'end'
     tabs_content:
       type: boolean
       label: 'Tabs content'
-      description: 'Display tabs content container'
+      description: 'If true, shows the tabs content container.'
```
Comment (abel-santos-corral) [inline templates/patterns/nav/nav.ui_patterns.yml]: Please, to homogenize, the formula 'When enable' could be used.

File: templates/patterns/nav/nav.ui_patterns.yml [hunk 3]
```diff
@@ -25,25 +25,25 @@ nav:
...
       preview: true
     nav:
       type: boolean
       label: 'Nav'
-      description: 'When enabled, use nav-style for show items instead unordered lists'
+      description: 'If true, use nav style for show items instead unordered lists.'
```
Comment (abel-santos-corral) [inline templates/patterns/nav/nav.ui_patterns.yml]: It seemed better as 'When enabled'.

Comment (abel-santos-corral) [inline templates/patterns/nav/nav.ui_patterns.yml]:
```suggestion
      description: 'When enabled, use navigation style for show items instead unordered lists.'
```

File: templates/patterns/offcanvas/offcanvas.ui_patterns.yml [hunk 4]
```diff
@@ -1,66 +1,57 @@
...
+        end: 'end'
+        top: 'top'
+        bottom: 'bottom'
+      preview: 'start'
     with_body_scroll:
-      type: "boolean"
-      label: "Enable body scroll when offcanvas is activated"
+      type: boolean
+      label: 'Scroll'
+      description: 'If true, enable body scrolling when offcanvas is displayed. False by default.'
```
Comment (abel-santos-corral) [inline templates/patterns/offcanvas/offcanvas.ui_patterns.yml]:
```suggestion
      description: 'When enabled, allows body scrolling when offcanvas is displayed. False by default.'
```

File: templates/patterns/offcanvas/offcanvas.ui_patterns.yml [hunk 5]
```diff
@@ -1,66 +1,57 @@
...
       preview: false
```
Comment (abel-santos-corral) [inline templates/patterns/offcanvas/offcanvas.ui_patterns.yml]: Not necessary, is false by default. To be checked at markup rendering test.

File: templates/patterns/offcanvas/offcanvas.ui_patterns.yml [hunk 6]
```diff
@@ -1,66 +1,57 @@
...
-      label: "Enable a button trigger for offcanvas"
+      type: boolean
+      label: 'Trigger'
+      description: 'If true, enable a button trigger for offcanvas'
       preview: true
     icon_position:
-      type: "select"
-      label: "Icon position"
+      type: select
+      label: 'Icon position'
```
Comment (abel-santos-corral) [inline templates/patterns/offcanvas/offcanvas.ui_patterns.yml]: Add description, please. Something as  'Sets the position of the icon regarding the text'. We must check, the icon position is in another item in another PR )this field is present at button, banner, link and off-canvas. We must ensure homogeneity between all patterns.

File: templates/patterns/pagination/pagination.ui_patterns.yml [hunk 7]
```diff
@@ -1,44 +1,50 @@
...
+      description: 'Align the pagination to:'
+      options:
+        center: 'center'
+        end: 'end'
+      preview: 'center'
     enable_icon:
       type: boolean
-      label: "Enable icon"
+      label: 'Enable icon'
+      description: 'Use icon instead of link labels to previous and next items.'
```
Comment (abel-santos-corral) [inline templates/patterns/pagination/pagination.ui_patterns.yml]: Add default value, please

File: templates/patterns/pagination/pagination.ui_patterns.yml [hunk 8]
```diff
@@ -1,44 +1,50 @@
...
       preview: false
```
Comment (abel-santos-corral) [inline templates/patterns/pagination/pagination.ui_patterns.yml]: Not necessary, its value is false by default. To be checked at markup test.

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 9]
```diff
@@ -1,49 +1,58 @@
...
+      type: boolean
+      label: 'Striped'
+      description: 'Stripped style for the progress bar. False by default.'
       preview: true
     animated:
-      type: "boolean"
-      label: "Animation on striped style of state"
+      type: boolean
+      label: 'Animated'
+      description: 'Animation on striped bar.'
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml]: Add value by default, please

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 10]
```diff
@@ -1,49 +1,58 @@
...
       preview: true
+    min:
+      type: number
+      label: 'Minimum'
+      description: 'Minimum state of progress bar. 0 by default. '
+      preview: 0
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml]: If default value is zero, not necessary

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 11]
```diff
@@ -1,49 +1,58 @@
...
-      type: "boolean"
-      label: "Animation on striped style of state"
+      type: boolean
+      label: 'Animated'
+      description: 'Animation on striped bar.'
       preview: true
+    min:
+      type: number
+      label: 'Minimum'
+      description: 'Minimum state of progress bar. 0 by default. '
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml]:
```suggestion
      description: 'Minimum state of progress bar. 0 by default.'
```

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 12]
```diff
@@ -1,49 +1,58 @@
...
+      label: 'Progress'
+      description: 'Progress of the bar.'
+      preview: 50
     bar_label:
-      type: "text"
-      label: "Text indicating the percentage of progress in the bar"
-      preview: "50%"
+      type: text
+      label: 'Bar label'
+      description: 'Text inside the progress bar'
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml]:
```suggestion
      description: 'Text inside the progress bar.'
```

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 13]
```diff
@@ -1,49 +1,58 @@
...
-      type: "text"
-      label: "Minimum state of prorgress bar"
-      preview: "0"
-    max:
-      type: "text"
-      label: "Maximum state of prorgress bar"
-      preview: "100"
+      type: text
+      label: 'Message'
+      description: 'Optional description text below the progress bar'
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml]:
```suggestion
      description: 'Optional description text below the progress bar.'
```

File: templates/patterns/table/table.ui_patterns.yml [hunk 14]
```diff
@@ -1,108 +1,118 @@
 table:
-  label: "Table"
-  description: "https://v5.getbootstrap.com/docs/5.0/content/typography/#table"
+  label: 'Table'
+  description: 'For more information: https://v5.getbootstrap.com/docs/5.0/content/typography/#table'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]:
```suggestion
  description: 'For more information visit: https://v5.getbootstrap.com/docs/5.0/content/typography/#table'
```

Comment (abel-santos-corral) [reply templates/patterns/table/table.ui_patterns.yml]: Maybe we can add some simple text before the part "For more info..."  just to say like 'Pattern to show a table'.

File: templates/patterns/table/table.ui_patterns.yml [hunk 15]
```diff
@@ -1,108 +1,118 @@
...
   settings:
     striped:
       type: boolean
-      label: Striped
+      label: 'Striped'
+      description: 'Add zebra-striping to any table row within the <tbody>.'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]: Add default value to description. Change to 'When enabled, adds a zebra-striping...'

File: templates/patterns/table/table.ui_patterns.yml [hunk 16]
```diff
@@ -1,108 +1,118 @@
...
     hoverable:
       type: boolean
-      label: Hoverable
+      label: 'Hoverable'
+      description: 'Enable a hover state on table rows within a <tbody>.'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]: Add default value to the description. Use other way of expressing: "When enabled, allows hover state..."

File: templates/patterns/table/table.ui_patterns.yml [hunk 17]
```diff
@@ -1,108 +1,118 @@
...
+        danger: 'danger'
+        warning: 'warning'
+        info: 'info'
+        light: 'light'
+        dark: 'dark'
     small:
       type: boolean
-      label: Small
+      label: 'Small'
+      description: 'If true, makes table more compact by cutting all cell padding in half. False by default.'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]:
```suggestion
      description: 'When enabled, makes table more compact by cutting all cell padding in half. False by default.'
```

File: templates/patterns/table/table.ui_patterns.yml [hunk 18]
```diff
@@ -1,108 +1,118 @@
...
+          - content: 'Name'
+          - content: 'Description'
+          - content: 'Position'
     table_footer:
-      type: "array"
-      label: table_footer
-      description: An array of a table footer items
+      type: array
+      label: 'Footer'
+      description: 'An array with the tfoot with items with content and col-span.'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]:
```suggestion
      description: 'An array with the table footer (tfoot) with items with content and col-span.'
```

File: templates/patterns/table/table.ui_patterns.yml [hunk 19]
```diff
@@ -1,108 +1,118 @@
...
+          - content: 'Name'
+          - content: 'Description'
+          - content: 'Position'
     rows:
-      type: "array"
-      label: rows
-      description: An array of a row items
+      type: array
+      label: 'Rows'
+      description: 'An array with the tbody with active, alignment, style and cells array with content and col-span.'
```
Comment (abel-santos-corral) [inline templates/patterns/table/table.ui_patterns.yml]:
```suggestion
      description: 'An array with the table body (tbody) with active, alignment, style and cells array with content and col-span.'
```

File: templates/patterns/timeline/timeline.ui_patterns.yml [hunk 20]
```diff
@@ -1,48 +1,50 @@
...
-      label: "Hide from"
-      description: "Number to hide 'from' items"
+      label: 'Hide from'
+      description: 'Indicate the number of items to display in the timeline, hide the others and add a toggle button to show them.'
       preview: 5
   fields:
     toggle_expanded:
       type: string
       label: 'Label expanded'
+      description: 'When all items are displaying, label of the button to hide them.'
```
Comment (abel-santos-corral) [inline templates/patterns/timeline/timeline.ui_patterns.yml]:
```suggestion
      description: 'When all items are displayed, the label of the button to hide them.'
```

File: templates/patterns/timeline/timeline.ui_patterns.yml [hunk 21]
```diff
@@ -1,48 +1,50 @@
...
     toggle_collapsed:
       type: string
       label: 'Label collapsed'
+      description: 'When there are hidden items, label of the button to expand them.'
```
Comment (abel-santos-corral) [inline templates/patterns/timeline/timeline.ui_patterns.yml]:
```suggestion
      description: 'When there are hidden items, the label of the button to expand them.'
```

**Review by abel-santos-corral — APPROVED** [review]

File: templates/patterns/nav/nav.ui_patterns.yml [hunk 22]
```diff
@@ -1,22 +1,21 @@
 nav:
   label: Navigation
-  description: 'https://v5.getbootstrap.com/docs/5.0/components/navs/'
```
Comment (brummbar) [inline templates/patterns/nav/nav.ui_patterns.yml:3]: We drop this pattern now.

File: templates/patterns/offcanvas/offcanvas.ui_patterns.yml [hunk 23]
```diff
@@ -1,66 +1,57 @@
 offcanvas:
-  label: "Offcanvas"
-  description: "Hidden sidebar with activation. https://getbootstrap.com/docs/5.0/components/offcanvas/"
+  label: 'Offcanvas'
+  description: 'Hidden sidebar with activation. For more information visit: https://getbootstrap.com/docs/5.0/components/offcanvas/'
```
Comment (brummbar) [inline templates/patterns/offcanvas/offcanvas.ui_patterns.yml:3]: Remove all the information about button and add a single "toggle" field. Type array, label "Toggle button", description "An optional button pattern definition that will allow to show/hide the offcanvas."
In the twig file we check. If the "toggle" variable is not empty, we merge a sensible default button with the values passed by the user.

We then do something like this:
```
{% if toggle is not empty %}
  {% set _toggle_attributes = toggle.attributes|default({}) %}
  {% set _toggle_attributes = _toggle_attributes|merge({
    'data-bs-toggle': 'offcanvas',
    'data-bs-target': '#' ~  offcanvas_id,
    'aria-controls': offcanvas_id
  }, true) %}
  {% set _toggle = toggle|merge({
    attributes: _toggle_attributes
  }) %}
  {{ pattern('button', _toggle) }}
{% endif %}
```

File: templates/patterns/pagination/pagination.ui_patterns.yml [hunk 24]
```diff
@@ -1,44 +1,49 @@
...
-      label: "Aria label"
-      preview: ""
+      type: select
+      label: 'Alignment'
+      description: 'Align the pagination to:'
+      options:
+        center: 'center'
+        end: 'end'
+      preview: 'center'
     enable_icon:
```
Comment (brummbar) [inline templates/patterns/pagination/pagination.ui_patterns.yml]: Not a super fan of this, but let's put at least `enable_icons` plural.

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 25]
```diff
@@ -1,49 +1,56 @@
...
+      description: 'Animation on striped bar. False by default.'
       preview: true
+    min:
+      type: number
+      label: 'Minimum'
+      description: 'Minimum state of progress bar. 0 by default.'
+    max:
+      type: number
+      label: 'Maximum'
+      description: 'Maximum state of progress bar. 100 by default.'
```
Comment (brummbar) [inline templates/patterns/progress/progress.ui_patterns.yml]: I would place these together with progress.

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 26]
```diff
@@ -1,49 +1,56 @@
...
+      preview: 'This is the label of the progress bar'
     progress:
-      type: "text"
-      label: "Percentage of progress in the bar"
-      preview: "50"
+      type: number
+      label: 'Progress'
+      description: 'Progress of the bar.'
+      preview: 50
     bar_label:
```
Comment (brummbar) [inline templates/patterns/progress/progress.ui_patterns.yml]: Maybe "text" for this?

File: templates/patterns/table/pattern-table.html.twig [hunk 27]
```diff
@@ -1,22 +1,29 @@
 {#
 /**
  * @file
- * Template for a Table component.
- *
+ * Table pattern.
```
Comment (brummbar) [inline templates/patterns/table/pattern-table.html.twig]: If not used in ucpkn, we drop this.

File: templates/patterns/timeline/pattern-timeline.html.twig [hunk 28]
```diff
@@ -8,13 +8,13 @@
 {% set _toggle_collapsed = toggle_collapsed|default('Show more %d items'|t) %}
 {% set _toggle_expanded = toggle_expanded|default('Hide %d items'|t) %}
 {% include '@oe-bcl/timeline' with {
-  toggle_collapsed: _toggle_collapsed,
-  toggle_expanded: _toggle_expanded,
-  hide: {
-    from: hide_from,
-    to: hide_to|default(0),
+  'toggle_collapsed': _toggle_collapsed,
+  'toggle_expanded': _toggle_expanded,
```
Comment (brummbar) [inline templates/patterns/timeline/pattern-timeline.html.twig]:
```suggestion
  'toggle_collapsed': toggle_collapsed|default('Show more %d items'|t) ,
  'toggle_expanded': toggle_expanded|default('Hide %d items'|t),
```
By the way, we need to add "label" to the name, or it seems a boolean variable. "collapse_button_label" and "expand_button_label" i think they are better.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: templates/patterns/nav/nav.ui_patterns.yml (hunk 22 above)
Comment (escuriola) [reply templates/patterns/nav/nav.ui_patterns.yml:3]: This pattern is being used in several places and in other patterns. Are you sure @brummbar

**Review by brummbar — APPROVED** [review]

**Pull request merged**